compiler/rustc_hir_analysis/src/variance/mod.rs

```diff
@@ -122,13 +122,6 @@ fn variance_of_opaque(tcx: TyCtxt<'_>, item_def_id: LocalDefId) -> &[ty::Varianc
                 {
                     self.visit_opaque(*def_id, substs)
                 }
-                // FIXME(-Zlower-impl-trait-in-trait-to-assoc-ty) check whether this is necessary
```

Comment:
Are you certain that this is correct to remove, or did you just remove this because no UI tests failed?

Reply:
Isn't it just equivalent to the previous arm?. Like in this new scheme, def_kind is going to be OpaqueTy so this branch is never gonna be exercised.

Reply:
Are you sure it's not being called on any of the GAT def ids?

Reply:
Ok, going to double check tomorrow. I guess you may be implying that for impls is executed because for traits shouldn't.

Reply:
In any case, given that all test pass here would be good to come up with a test case that fails with this branch removed and works if we add this back.

Reply:
For traits, walking the fn_sig will give you back projections, so it may happen. Yeah, this at least needs a bit more checking.